pypom/page.py

```diff
@@ -2,45 +2,81 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+from .exception import UsageError
 from .view import WebView
 
 
 class Page(WebView):
 
-    _url = '{base_url}'
+    URL_TEMPLATE = ''
     """String representing a URL that will return this page.
 
     This string is formatted and can contain names of keyword arguments passed
-    during construcion of the page object. The ``{base_url}`` keyword argument
-    is always passed to the format operation.
+    during construction of the page object. The template should either assume
+    that its result will be appended to the value of ``base_url``,
+    or should yield an absolute URL.
     """
 
-    def open(self):
+    @property
+    def canonical_url(self):
         """
-        Navigates to the URL returned by the :py:func:`url` property and
-        waits for the page to load by calling :py:func:`wait_for_page_to_load`.
+        Returns the URL to the current page,
+        formatted from :py:data:`URL_TEMPLATE`.
+        Unless URL_TEMPLATE results in an absolute URL, ``self.base_url`` is
+        prepended to the resulting URL.
 
         :returns:
-            The current page object (i.e., ``self``).
+            String representing a URL that will return this page.
         """
-        self.selenium.get(self.url)
-        self.wait_for_page_to_load()
-        return self
+        url = self.URL_TEMPLATE.format(**self.url_kwargs)
+        return ('//' in url and url) or (self.base_url and self.base_url + url)
```

Comment:
Not sure I understand this. Doesn't this mean a URL like `/path//path` wouldn't be appended to the base URL? I know it's unlikely but it's a valid URL path.